src/Commands/TerminusCommand.php

```diff
@@ -2,34 +2,36 @@
 
 namespace Pantheon\Terminus\Commands;
 
+use Pantheon\Terminus\Config;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
 use Symfony\Component\Console\Input\InputAwareInterface;
 use Robo\Contract\OutputAwareInterface;
 use Robo\Common\OutputAwareTrait;
 use Robo\Common\InputAwareTrait;
 use Terminus\Models\Auth;
 
-abstract class TerminusCommand implements InputAwareInterface, OutputAwareInterface
+abstract class TerminusCommand implements InputAwareInterface, LoggerAwareInterface, OutputAwareInterface
 {
     use InputAwareTrait;
+    use LoggerAwareTrait;
     use OutputAwareTrait;
 
     /**
-     * @var Config
-     */
-    protected $config;
-    /**
      * @var boolean True if the command requires the user to be logged in
      */
     protected $authorized = false;
+    /**
+     * @var Config
+     */
+    protected $config;
 
     /**
      * TerminusCommand constructor
-     *
-     * @param Config $config Terminus configuration object
      */
-    public function __construct($config)
+    public function __construct()
     {
-        $this->config = $config;
+        $this->config = new Config();
```

Comment:
It's really not that hard. Robo's injecting magic is designed to be extensible.

1. Make yourself a new TerminusConfigAwareInterface / Trait.
2. Make TerminusCommand implement the interface and use the trait.
3. Share your Config singleton in the container.
4. Add an inflector to the league/container at startup to set your config object on anything that implements TerminusConfigAwareInterface.

For that matter, you might be able to just get away with using Robo's config object. It has getters and setters; might work.  Or, subclass it and use Robo's ConfigAwareInterface / ConfigAwareTrait.

Yes, Robo rocks.  I'll show you how to do (3) and (4) tomorrow. It's just a few easy lines of code, because league/container rocks.  It's a pity that so many people try to learn DI with the Symfony container, because I agree that Symfony service containers are prone to making heads hurt. But it doesn't have to be painful.